merlin/dataloader/torch.py

```diff
@@ -141,7 +141,7 @@ def _row_lengths_to_offsets(self, row_lengths):
         zero_value = torch.tensor([0], device=self.device, dtype=row_lengths.dtype)
         if len(row_lengths.shape) == 2:
             zero_value = zero_value.view(-1, 1)
-        return torch.cat((zero_value, torch.cumsum(row_lengths, 0)))
+        return torch.cat((zero_value, torch.cumsum(row_lengths, 0).to(device=self.device)))
```

Comment:
assuming `torch.cat` preserves the device of the values being concatented, maybe an alternative to this would be to change the line where `zero_value` is defined to use `row_lengths.device` instead of `self.device`?

Reply:
Thanks for the suggestion. Honestly I'm not sure which option is best, but it also works, and I think your suggestion is better. Changed in [de593b2](https://github.com/NVIDIA-Merlin/dataloader/pull/113/commits/de593b2b25117d72078463e6745402d8510dc014).